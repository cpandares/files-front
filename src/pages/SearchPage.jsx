
import { useLocation, useNavigate } from "react-router-dom";
import queryString from "query-string";
import { useForm } from "../hooks/useForm";

import useFiles from "../hooks/useFiles";
import TableData from "../components/TableData";


const SearchPage = () => {


  const navigate = useNavigate();
  const location = useLocation();

  const { q = "" } = queryString.parse(location.search);

  const { file } = useFiles(q);  

  const showSearch = q.length === 0;
  

  const { searchText, onInputChange } = useForm({
    searchText: q,
  });

  const handleSubmit = (e) => {
    e.preventDefault();
    if (searchText.trim().length <= 1) return;

    navigate(`?q=${searchText}`);
  };

  return (
    <div className="row container mt-5">

      <div className="col-5">

        <form 
          onSubmit={handleSubmit}
          aria-label="form"
          >
          <input
            type="text"
            className="form-control"
            placeholder="Example: test1.csv"
            name="searchText"
            value={searchText}
            onChange={onInputChange}
          />

          <button className="mt-2 btn btn-outline-primary">Search</button>
        </form>
      </div>

      <div className="col-7">
        <h4>Results</h4>
        <hr />

        <div
          className="alert alert-primary"
          aria-label="alert-primary"
          style={{ display: showSearch ? "" : "none" }}
        >
          Search a File
        </div>      
         

        <TableData file = { file } />
       
      </div>
    </div>
  );
};

export default SearchPage;
