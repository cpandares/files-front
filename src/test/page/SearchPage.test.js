
import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter }   from 'react-router-dom';
import SearchPage from '../../pages/SearchPage';



const mockedUseNavigate = jest.fn();

jest.mock("react-router-dom", ()=> ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: ()=>mockedUseNavigate

}) );

describe("Pruebas en el <SearchPage />", ()=>{

    beforeEach(()=> jest.clearAllMocks())

    test("Debe de llar el navigate a la nueva pantalla", ()=>{

        render(
            <MemoryRouter initialEntries={['/search?q=test3.csv']}>
                <SearchPage />
            </MemoryRouter>
        );


        const input = screen.getByRole('textbox');
        expect( input.value ).toBe('test3.csv');

        

        const form = screen.getByRole('form');
        fireEvent.submit(form);

        expect( mockedUseNavigate ).toBeCalledWith('?q=test3.csv');

    
    });
    
    test("Debe de mostar el componente con valores por defecto",()=>{

        const { container } =  render(
               <MemoryRouter>
                   <SearchPage />
               </MemoryRouter>
           );
   
           expect( container ).toMatchSnapshot()
   
   
    })


    test("Debe Ocultar el ' Search a File' si el input contiene un texto", ()=>{

        render(
            <MemoryRouter initialEntries={['/search?q=test3.csv']}>
                <SearchPage />
            </MemoryRouter>
        );


        const input = screen.getByRole('textbox');
        expect( input.value ).toBe('test3.csv');

        

        const div = screen.getByLabelText('alert-primary');
        expect(div.style.display).toBe('none');

    
    });

   

    test("Debe mostrar un error si no encuentra el file", ()=>{

        render(
            <MemoryRouter initialEntries={['/search?q=test7.csv']}>
                <SearchPage />
            </MemoryRouter>

        )

        const div = screen.getByLabelText('no-results');
        expect(div).toBeVisible("<td aria-label='no-results' colspan='4'>No Results</td>")

    })


})