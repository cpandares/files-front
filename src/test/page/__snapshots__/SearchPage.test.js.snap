// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <SearchPage /> Debe de mostar el componente con valores por defecto 1`] = `
<div>
  <div
    class="row container mt-5"
  >
    <div
      class="col-5"
    >
      <form
        aria-label="form"
      >
        <input
          class="form-control"
          name="searchText"
          placeholder="Example: test1.csv"
          type="text"
          value=""
        />
        <button
          class="mt-2 btn btn-outline-primary"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        aria-label="alert-primary"
        class="alert alert-primary"
      >
        Search a File
      </div>
      <table
        class="table table-striped table-bordered container mt-4 text-center "
      >
        <thead>
          <tr>
            <th>
              FileName
            </th>
            <th>
              Text
            </th>
            <th>
              Number
            </th>
            <th>
              Hex
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              aria-label="no-results"
              colspan="4"
            >
              No Results
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
